modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,208 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-kuryr-sdn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-kuryr-cleanup_{context}"]
+= Cleaning up resources after migration
+
+After migration is done, there are necessary steps to clean up resources used by Kuryr. From high level perspective it will include:
+
+* Removing Kuryr finalizers from resources:
+* Removing Kuryr CRDs
+* Removing {rh-openstack-first} resources
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install `jq` tool
+* Install Python interpreter
+* Install `openstacksdk` python package
+* Install `openstack` cli tool
+* Access to the underlying {rh-openstack}
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Find out cluster and Kuryr related identifiers, which will be used later:
++
+[source,terminal]
+----
+$ CLUSTERID=$(oc get infrastructure.config.openshift.io cluster -o=jsonpath='{.status.infrastructureName}')
+$ CLUSTERTAG="openshiftClusterID=${CLUSTERID}"
+$ ROUTERID=$(oc get kuryrnetwork -A --no-headers -o custom-columns=":status.routerId"|head -n 1)
+----
+
+. Create helper bash function, which will help out with removing finalizers
+from specified resources. It might look complicated, but it simply gets two
+parameters - first one is name of the resource, and the second is the
+finalizer to be removed. Than it will get all the desired resource from the
+cluster and replace it's definition with the same data excluding specified
+finalizer.
++
+[source,terminal]
+----
+$ function REMFIN {
+    local resource=$1
+    local finalizer=$2
+    for res in $(oc get $resource -A --template='{{range $i,$p := .items}}{{ $p.metadata.name }}|{{ $p.metadata.namespace }}{{"\n"}}{{end}}'); do
+        name=${res%%|*}
+        ns=${res##*|}
+        oc get -n $ns $resource $name -o json |
+            jq -Mcr "if .metadata.finalizers != null then del(.metadata.finalizers[] | select(. == \"${finalizer}\")) else . end" |
+            oc replace -n $ns $resource $name -f -
```

Comment:
We could have it retry on conflict, but in general just running it again is safe. We should just mention that it's idempotent and you should try again on conflict errors.